src/os/bluestore/BlueFS.h

```diff
@@ -344,13 +344,15 @@ class BlueFS {
   uint64_t _default_bluefs_alloc_size(uint64_t bluestore_min_alloc_size) {
     uint64_t min_allowed_size = 16384U; // same as bluestore_min_alloc_size_ssd
     if (cct->_conf->bluefs_alloc_size > 0 &&
-	  cct->_conf->bluefs_alloc_size >= min_allowed_size)
+	  cct->_conf->bluefs_alloc_size >= min_allowed_size) {
+      if (bluestore_min_alloc_size > 0)
+        ceph_assert(cct->_conf->bluefs_alloc_size % bluestore_min_alloc_size == 0);
       return cct->_conf->bluefs_alloc_size;
-    else if (bluestore_min_alloc_size > 0 &&
+    } else if (bluestore_min_alloc_size > 0 &&
 	       bluestore_min_alloc_size >= min_allowed_size)
-      return bluestore_min_alloc_size;
-    else
-      return min_allowed_size;
+        return bluestore_min_alloc_size;
+      else
+        return min_allowed_size;
```

Comment:
may be something like the following instead?

uint64_t _default_bluefs_alloc_size(uint64_t bluestore_min_alloc_size) {
  if (cct->_conf->bluefs_alloc_size > 0) {
      if (bluestore_min_alloc_size > 0)
        ceph_assert(cct->_conf->bluefs_alloc_size % bluestore_min_alloc_size == 0);
      return cct->_conf->bluefs_alloc_size;
  }
  return bluestore_min_alloc_size;
}

Reply:
@ifed01 this will make `bluefs_alloc_size`= 0 for the bluestore_tool and the tests, is that acceptable?

Reply:
It should be... The cases when we pass 0 to BlueFS ctor don't need this value - they use bluefs in read-only mode. Also you made an assertion in BlueFS::_init_alloc that protects from accidental use of bluefs_alloc_size=0 if any. So hopefully we're safe.

Reply:
@ifed01 this is making the unittests in test/objectstore/test_bluefs.cc fail with `/home/ubuntu/ceph/src/os/bluestore/BlueFS.cc: 444: FAILED ceph_assert(bluefs_alloc_size != 0)`, should we use a non-zero value for the unittests, if so, what should be that value?

Reply:
how about we do something like https://github.com/ceph/ceph/blob/master/src/test/objectstore/test_bluefs.cc#L305-L308 and pass bluefs_alloc_size?

Reply:
As they were originally intended for 1M it makes sense to use 1Mb as the first step.
Additionally you might want to use various alloc_size values there, e.g. 16K and 64K are the best candidates since these ones are going to be used by default. 4K might be a good choice too. 
You may use src/test/compressor/test_compression.cc as an example how to make parameterized tests.

Reply:
Wherever required I have used 1Mb for now, some tests are already using 64K. We can convert them to parameterized tests in follow up PR.